test/e2e/specs/editor/various/change-detection.spec.js

```diff
@@ -0,0 +1,553 @@
+			.click();
+
+		await expect(
+			page
+				.getByRole( 'region', { name: 'Editor top bar' } )
+				.getByRole( 'button', { name: 'saved' } )
+		).toBeDisabled();
+
+		await expect( page ).toHaveURL( '/wp-admin/edit.php?post_type=post' );
+	} );
+
+	test( 'consecutive edits to the same attribute should mark the post as dirty after a save', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		// Open the sidebar block settings.
+		await editor.openDocumentSettingsSidebar();
+
+		// Insert a paragraph.
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( 'Hello, World!' );
+
+		// Save and wait till the post is clean.
+		await Promise.all( [
+			expect(
+				page
+					.getByRole( 'region', { name: 'Editor top bar' } )
+					.getByRole( 'button', { name: 'saved' } )
+			).toBeDisabled(),
+			pageUtils.pressKeys( 'primary+s' ),
+		] );
+
+		// Change the paragraph's `drop cap`.
+		await page
+			.getByRole( 'region', { name: 'Editor settings' } )
+			.getByRole( 'button', { name: 'Typography options' } )
+			.click();
+		await page
+			.getByRole( 'menu', { name: 'Typography options' } )
+			.getByRole( 'menuitemcheckbox', { name: 'Show drop cap' } )
+			.click();
+
+		await page
+			.getByRole( 'region', { name: 'Editor settings' } )
+			.getByRole( 'checkbox', { name: 'Drop cap' } )
+			.setChecked( true );
+
+		// Check that the post is dirty.
+		await expect(
+			page
+				.getByRole( 'region', { name: 'Editor top bar' } )
+				.getByRole( 'button', { name: 'Save draft' } )
+		).toBeEnabled();
+
+		// Save and wait till the post is clean.
+		await Promise.all( [
+			expect(
+				page
+					.getByRole( 'region', { name: 'Editor top bar' } )
+					.getByRole( 'button', { name: 'saved' } )
+			).toBeDisabled(),
+			pageUtils.pressKeys( 'primary+s' ),
+		] );
+
+		// Change the paragraph's `drop cap` again.
+		await page
+			.getByRole( 'region', { name: 'Editor settings' } )
+			.getByRole( 'checkbox', { name: 'Drop cap' } )
+			.setChecked( false );
+
+		// Check that the post is dirty.
+		await expect(
+			page
+				.getByRole( 'region', { name: 'Editor top bar' } )
+				.getByRole( 'button', { name: 'Save draft' } )
+		).toBeEnabled();
+	} );
+} );
+
+class ChangeDetectionUtils {
+	/** @type {Page} */
+	#page;
+	/** @type {(() => void) | null} */
+	#continueInterceptedSave = null;
+
+	constructor( { page } ) {
+		this.#page = page;
+	}
+
+	/**
+	 * @return {boolean} Whether the post is dirty.
+	 */
+	getIsDirty = async () => {
+		return await test.step(
+			'assert is dirty',
+			async () => {
+				const hasDialog = new Promise( ( resolve ) => {
+					this.#page.on( 'dialog', ( dialog ) => {
+						void dialog.accept();
+						resolve( true );
+					} );
+					this.#page.on( 'load', () => resolve( false ) );
+				} );
+				await this.#page.reload( { waitUntil: 'commit' } ); // No need to wait for the full load.
+				return hasDialog;
```

Comment:
This is very interesting. Based on docs, PW only handles `beforeunload` dialogs when calling `page.close({ runBeforeUnload: true })`. They aren't run on `page.reload`, so I'm really curious how this assertion is working 😅

Reply:
I couldn't find the part where the doc is referencing that though. I think the doc means that it won't wait for the `beforeunload` dialogs unless `runBeforeUnload` is true. But that's just for `page.close()`. `beforeunload` dialogs should still be available in the event handler, but since we don't care whether Playwright waits for the dialog to be closed or not, I think this is fine?

Reply:
Here's the link - https://playwright.dev/docs/dialogs#beforeunload-dialog. You're right; this doc probably refers to page leave context.

 > I think this is fine?

As long as assertions work, it's golden 😄